### PR TITLE
Message hardening

### PR DESCRIPTION
**Pull request by lisadier**
Updates made to clean up FHIRConverterTest, and other hardening to remove disabled tests and some TODOs that could be updated, as well as additional coverage.

### CODE REVIEW

File: src/main/resources/hl7/message/ADT_A08.yml [hunk 1]
```diff
@@ -12,48 +12,60 @@ resources:
...
-             - MSH
+        - MSH
+
+    - resourceName: Procedure
+      segment: .PR1
+      group: PROCEDURE
+      resourcePath: resource/Procedure
+      repeats: true
+      additionalSegments:
+        - .ROL
```
Comment (klwhaley) [inline src/main/resources/hl7/message/ADT_A08.yml:68]: do we have any support for ROL yet?  maybe thats something done earlier?

File: src/test/java/io/github/linuxforhealth/hl7/message/HL7ADTMessageTest.java [hunk 2]
```diff
@@ -28,403 +30,525 @@
...
+                        .filter(v -> ResourceType.AllergyIntolerance == v.getResource().getResourceType())
+                        .map(BundleEntryComponent::getResource).collect(Collectors.toList());
+        assertThat(allergyIntoleranceResource).hasSize(2); // from AL1
+
+        List<Resource> conditionResource = e.stream()
+                        .filter(v -> ResourceType.Condition == v.getResource().getResourceType())
+                        .map(BundleEntryComponent::getResource).collect(Collectors.toList());
+        assertThat(conditionResource).hasSize(1); // from DG1
+
+        // Confirm that there are no extra resources
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/message/HL7ADTMessageTest.java:166]: should we add a check for document reference and set to 0, but that it should be 1 wiht the content from OBX 'tx' and reference the card?

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java [hunk 3]
```diff
@@ -247,102 +216,6 @@ public void test_adt_40_message_with_adt_a39_structure_specified() throws Except
 
   }
 
-  @Test
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:250]: I thought the point of this test was to make sure a message with no msh 9.3 was parse and converted due to a old bug.  Do we have anohter test for that?

**Review by klwhaley — APPROVED** [review]
Approved, with a few questions to consider

File: src/main/resources/hl7/message/ADT_A08.yml (hunk 1 above)
Comment (lisadier) [reply src/main/resources/hl7/message/ADT_A08.yml:68]: Right, we haven't done any work for ROL, this is the same way it was done in ADT_A01.  It does match the grouping and definition of segments for the message, however it is not currently used in Procedure.yml.  I could go either way on this one.  It isn't wrong yet it isn't yet used in Procedure.  How about we leave it for now, and adjust if needed when we implement PR1 (for Procedures).

File: src/test/java/io/github/linuxforhealth/hl7/message/HL7ADTMessageTest.java (hunk 2 above)
Comment (lisadier) [reply src/test/java/io/github/linuxforhealth/hl7/message/HL7ADTMessageTest.java:166]: I agree, good idea, done.

File: src/test/java/io/github/linuxforhealth/FHIRConverterTest.java (hunk 3 above)
Comment (lisadier) [reply src/test/java/io/github/linuxforhealth/FHIRConverterTest.java:250]: Good catch, I didn't notice that was the point of the test.  I've added more comments to make it more clear why this test is here.

**Pull request merged**